packages/app/obojobo-repository/shared/components/notification.test.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react'
+import { create, act } from 'react-test-renderer'
+import Notification from './notification'
+
+describe('Notification component', () => {
+	beforeAll(() => {
+		Object.defineProperty(document, 'cookie', {
+			value: '',
+			writable: true
+		})
+	})
+
+	test('renders without crashing', () => {
+		const component = create(<Notification onDataFromNotification={() => {}} />) // Provide a mock function for onDataFromNotification
+		const tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+	})
+
+	test('loads notifications from cookies on mount', () => {
+		const onDataFromNotification = jest.fn()
+		document.cookie =
+			'notifications=' +
+			JSON.stringify([{ key: 1, text: 'Test Notification', title: 'Test Title' }])
+
+		const component = create(<Notification onDataFromNotification={onDataFromNotification} />)
+		const tree = component.toJSON()
+
+		expect(tree).toMatchSnapshot()
+		expect(document.cookie).toBe(
+			'notifications=%5B%7B%22key%22%3A1%2C%22text%22%3A%22Test%20Notification%22%2C%22title%22%3A%22Test%20Title%22%7D%5D;'
```

Comment:
This seems a bit magical, you should be able to store the important value in a variable and use `encodeURIComponent` to check it:
```javascript
const notificationValue = [{ key: 1, text: 'Test Notification', title: 'Test Title' }]

document.cookie = `notifications=${JSON.stringify(notificationValue)}`

...

expect(document.cookie.toBe(`notifications=${encodeURIComponent(notificationValue}`)
```

Reply:
`expect(document.cookie).toBe(`notifications=${encodeURIComponent(JSON.stringify(notificationValue))};`)`
good idea: had to stringify before encoding or it didn't catch the specific title/text